bin/runtime-common/src/messages.rs

```diff
@@ -326,8 +330,17 @@ pub mod source {
 		relayer_fee_percent: u32,
 	) -> Result<BalanceOf<ThisChain<B>>, &'static str> {
 		// the fee (in Bridged tokens) of all transactions that are made on the Bridged chain
-		let delivery_transaction =
-			BridgedChain::<B>::estimate_delivery_transaction(&payload.call, payload.weight.into());
+		//
+		// if we're going to pay dispatch fee at the target chain, then we don't include weight
+		// of the message dispatch in the delivery transaction cost
+		let delivery_transaction = BridgedChain::<B>::estimate_delivery_transaction(
+			&payload.call,
+			if payload.pay_dispatch_fee_at_target_chain {
+				0.into()
```

Comment:
I wonder if we should include the (potentially failed) transfer cost or emitting even from dispatch pallet here.

Reply:
This is only about message dispatch weight. All extra costs related to dispatch are already included in the delivery tx weight.